browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc

```diff
@@ -101,121 +84,22 @@ AIChatUIPageHandler::AIChatUIPageHandler(
 
 AIChatUIPageHandler::~AIChatUIPageHandler() = default;
 
-void AIChatUIPageHandler::SetClientPage(
-    mojo::PendingRemote<ai_chat::mojom::ChatUIPage> page) {
-  page_.Bind(std::move(page));
-
-  // In some cases, this page handler hasn't been created and remote might not
-  // have been set yet.
-  // ex. A user may ask a question from the location bar
-  if (active_chat_tab_helper_ &&
-      active_chat_tab_helper_->HasPendingConversationEntry()) {
-    OnHistoryUpdate();
-  }
-}
-
-void AIChatUIPageHandler::GetModels(GetModelsCallback callback) {
-  if (!active_chat_tab_helper_) {
-    VLOG(2) << "Chat tab helper is not set";
-    std::move(callback).Run(std::vector<mojom::ModelPtr>(), std::string());
-    return;
-  }
-
-  const auto& models = active_chat_tab_helper_->GetModels();
-  std::vector<mojom::ModelPtr> models_copy(models.size());
-  std::transform(models.cbegin(), models.cend(), models_copy.begin(),
-                 [](auto& model) { return model.Clone(); });
-
-  std::move(callback).Run(std::move(models_copy),
-                          active_chat_tab_helper_->GetCurrentModel().key);
-}
-
-void AIChatUIPageHandler::ChangeModel(const std::string& model_key) {
-  active_chat_tab_helper_->ChangeModel(model_key);
-}
-
-void AIChatUIPageHandler::SubmitHumanConversationEntry(
-    const std::string& input) {
-  DCHECK(!active_chat_tab_helper_->IsRequestInProgress())
-      << "Should not be able to submit more"
-      << "than a single human conversation turn at a time.";
-
-  mojom::ConversationTurnPtr turn = mojom::ConversationTurn::New(
-      CharacterType::HUMAN, mojom::ActionType::UNSPECIFIED,
-      ConversationTurnVisibility::VISIBLE, input, std::nullopt, std::nullopt,
-      base::Time::Now(), std::nullopt, false);
-  active_chat_tab_helper_->SubmitHumanConversationEntry(std::move(turn));
-}
-
-void AIChatUIPageHandler::SubmitHumanConversationEntryWithAction(
-    const std::string& input,
-    mojom::ActionType action_type) {
-  DCHECK(!active_chat_tab_helper_->IsRequestInProgress())
-      << "Should not be able to submit more"
-      << "than a single human conversation turn at a time.";
-
-  active_chat_tab_helper_->SubmitSelectedText(input, action_type);
-}
-
-void AIChatUIPageHandler::SubmitSummarizationRequest() {
-  if (active_chat_tab_helper_) {
-    active_chat_tab_helper_->SubmitSummarizationRequest();
-  }
-}
-
 void AIChatUIPageHandler::HandleVoiceRecognition() {
 #if BUILDFLAG(IS_ANDROID)
+  // TODO(petemill): figure out why java needs access to conversation context
+  // web contents and this can't be done in UI WebContents only. Taget
+  // WebContents
   auto* context_web_contents = (active_chat_tab_helper_)
```

Comment:
@SergeyZhukovsky can this just always be the WebUI's `web_contents()`?

Reply:
it used to pass a Leo query after voice recognition work https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/brave_leo/BraveLeoVoiceRecognitionHandler.java#L69.

Reply:
ah ok thanks. In this refactor, a Conversation might not have a WebContents context, since it might be standalone. I think we could change this to pass a simple string conversation id?

Reply:
@SergeyZhukovsky I have pushed an android change to target a specific conversation when calling `handleVoiceRecognition`. Please take a look 7e81dc4b1e4a6d38f4152b348c051ccda49e34dd